internal/dependencymanager/dependencyinstaller.go

```diff
@@ -198,12 +206,47 @@ func (di *DependencyInstaller) Add(depSource, customName string) error {
 	return nil
 }
 
+func (di *DependencyInstaller) addDependency(dep config.Dependency) error {
+	if _, exists := di.dependencies[dep.Source.Address.String()]; exists {
+		return nil
+	}
+
+	di.dependencies[dep.Source.Address.String()] = dep
+
+	return nil
+
+}
+
+func (di *DependencyInstaller) handleClosingTasks() {
```

Comment:
what's this function do? I'm not sure if the function name "handlclosingTasks" is a good name, we could add some comments if you believe otherwise!

Reply:
Yeah I don't love the name either. Changed.